force-app/main/default/classes/ADDR_Addresses_UTIL_TEST.cls

```diff
@@ -0,0 +1,1341 @@
+        System.assertEquals(addressSource.MailingCity__c, addressDestination.MailingCity__c);
+        System.assertEquals(addressSource.MailingCounty__c, addressDestination.MailingCounty__c);
+        System.assertEquals(addressSource.MailingState__c, addressDestination.MailingState__c);
+        System.assertEquals(addressSource.MailingPostalCode__c, addressDestination.MailingPostalCode__c);
+        System.assertEquals(addressSource.MailingCountry__c, addressDestination.MailingCountry__c);
+        System.assertEquals(addressSource.Geolocation__Latitude__s, addressDestination.Geolocation__Latitude__s);
+        System.assertEquals(addressSource.Geolocation__Longitude__s, addressDestination.Geolocation__Longitude__s);
+        System.assertEquals('Home', addressDestination.Address_Type__c);
+    }
+
+    @isTest
+    static void testCopyAddressObjNotNullSourceTypeBlank() {
+        Address__c addressSource = UTIL_UnitTestData_TEST.getMultipleTestAddresses(1)[0];
+        Address__c addressDestination = new Address__c();
+        addressSource.Address_Type__c = '';
+        addressDestination.Address_Type__c = 'Work';
+
+        Test.startTest();
+        ADDR_Addresses_UTIL.copyAddressObj(addressSource, addressDestination);
+        Test.stopTest();
+
+        System.assertEquals(addressSource.Default_Address__c, addressDestination.Default_Address__c);
+        System.assertEquals(addressSource.MailingStreet__c, addressDestination.MailingStreet__c);
+        System.assertEquals(addressSource.MailingStreet2__c, addressDestination.MailingStreet2__c);
+        System.assertEquals(addressSource.MailingCity__c, addressDestination.MailingCity__c);
+        System.assertEquals(addressSource.MailingCounty__c, addressDestination.MailingCounty__c);
+        System.assertEquals(addressSource.MailingState__c, addressDestination.MailingState__c);
+        System.assertEquals(addressSource.MailingPostalCode__c, addressDestination.MailingPostalCode__c);
+        System.assertEquals(addressSource.MailingCountry__c, addressDestination.MailingCountry__c);
+        System.assertEquals(addressSource.Geolocation__Latitude__s, addressDestination.Geolocation__Latitude__s);
+        System.assertEquals(addressSource.Geolocation__Longitude__s, addressDestination.Geolocation__Longitude__s);
+        System.assertEquals('Work', addressDestination.Address_Type__c);
+    }
+
+    /**************************************************************************************************************************
+     ****************************************************** copyAddressStdSObj ************************************************
+     **************************************************************************************************************************/
+    @isTest
+    static void testCopyAddressObjSobjSourceRecordNull() {
+        //To Refactor: we don't have guard clause for this.
```

Comment:
Regardless, capture the expected behavior and assert for it to ensure future refactors work as expected. Then make a note in the method. Also about the terrible method name.